projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-util.ts

```diff
@@ -97,12 +98,12 @@ export class PivotUtil {
                 expansionStates.get(expansionRowKey);
             if (rec[field + '_' + pivotKeys.records] &&
                 rec[field + '_' + pivotKeys.records].length > 0 &&
-                isExpanded && lvl > 0) {
+                ((isExpanded && lvl > 0)  || (maxDimLvl == currDimLvl))) {
```

Comment:
More specifically this `lvl > 0`

Reply:
Should be fine with just the new level check but I'll test it to make sure.